### DISCUSSION TITLE
Explain the difference between fixupCorridor and dtMergeCorridorStartMoved functions

### DISCUSSION THREAD

**Discussion by elsid** [recastnavigation/recastnavigation#633] [Q&A] [answered]
The `fixupCorridor` function is defined [here](https://github.com/recastnavigation/recastnavigation/blob/6dc1667f580357e8a2154c28b7867bea7e8ad3a7/RecastDemo/Source/NavMeshTesterTool.cpp#L63-L106) as a part of RecastDemo tool and [dtMergeCorridorStartMoved](https://github.com/recastnavigation/recastnavigation/blob/6dc1667f580357e8a2154c28b7867bea7e8ad3a7/DetourCrowd/Source/DetourPathCorridor.cpp#L27-L70) is a part of DetourCrowd. The functions are identical from semantic point of view. The only difference in code is use of `rcMin` vs `dtMin` and `rcMax` vs `dtMax` functions. Why not use `dtMergeCorridorStartMoved` instead of `fixupCorridor` and remove `fixupCorridor` to avoid code duplication?

**Comment by grahamboree** [comment, marked as answer]
Thanks for pointing this out!  It's likely that this was an oversight and RecastDemo should be using `dtMergeCorridorStartMoved` rather than re-implementing it.  I've added this as an issue: #634 

For context, we try to make the coupling of recast and detour as minimal as possible.  The idea is that you should (in theory at least) be able to use either independent of the other.  As a result, there are a handful of helper functions that are duplicated in both recast and detour and only differ by their `rc` or `dt` prefix.
